src/platform/packages/shared/kbn-esql-validation-autocomplete/src/autocomplete/helper.ts

```diff
@@ -1038,3 +1044,118 @@ export function isExpressionComplete(
     !(isNullMatcher.test(innerText) || isNotNullMatcher.test(innerText))
   );
 }
+
+export function getSourceSuggestions(sources: ESQLSourceResult[]) {
+  // hide indexes that start with .
+  return buildSourcesDefinitions(
+    sources
+      .filter(({ hidden }) => !hidden)
+      .map(({ name, dataStreams, title, type }) => {
+        return { name, isIntegration: Boolean(dataStreams && dataStreams.length), title, type };
+      })
+  );
+}
```

Comment:
Should this logic just move to `buildSourcesDefinitions`?

Reply:
I didnt do any changes to the existing logic. I left it as it is and is not part of this PR to refactor